File: api/src/Domain/ProductCatalog/PadSeries/DataTransferObjects/PadCreateOrUpdate.cs
namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadCreateOrUpdate : IDataTransferObject {
        public string Name { get; }
        public PadCategory Category { get; }
        public DataUrlImage? Image { get; }

        public PadCreateOrUpdate(string name, PadCategory category, DataUrlImage? image) {
            Name = name;
            Category = category;
            Image = image;
        }
    }
}

File: app/src/api/product-catalog/pad-series/data-transfer-objects/requests/pad-create-or-update.ts
import { PadCategory } from '@/api';
import { Image } from '@/api/product-catalog/common/data-transfer-objects/image';
import { PadMaterial } from '../pad-material';
import { PadSeriesSize } from '../pad-series-size';
import { PadTexture } from '../pad-texture';
import { PolisherType } from '../polisher-type';

export interface PadCreateOrUpdate {
    name: string,
    category: PadCategory,
    material: PadMaterial,
    texture: PadTexture,
    polisherTypes: PolisherType[],
    image: Image
}